### PR TITLE
[READY] Fixes megafauna deathgasp and hierophants death animation

### PR DESCRIPTION
**Pull request by Whoneedspacee**
## About The Pull Request

Increases the volume of death sounds, fixes hierophants ending animation not playing correctly caused by #42290 and fixes death sounds AND messages not playing for del_on_death creatures (most megafauna death sounds and death messages were broken by this) caused by #41493. Also adds a force achievement option where the megafauna can force the people that are selected to be granted the medal instead of it just being in a range.

## Why It's Good For The Game

Megafauna dying was less cool because of this, especially hierophant. There's lots of neat lore and flavortext in some of the deaths as well and it just feels kind of empty without the death sounds and messages, they just disappear.

## Changelog
:cl: Joe Berry
fix: Fixes various mobs death sounds and messages not functioning
fix: Fixes hierophants death animation not working properly
/:cl:

### CODE REVIEW

**Comment by 4dplanner** [comment]
The reason I let hierophant die the way it did was because of the text it gives- [src] shrinks, releasing a massive burst of energy! To me, at least, this suggested the shrinking and the burst happened at the same time, while the previous animation was burst->die after burst is finished

**Comment by Whoneedspacee** [comment]
@4dplanner Fair enough. Once I fixed the death sound being broken though it felt weird to me instantly going off instead of on a delay.

The old old death animation was a really cool part of hierophant anyways, you don’t really see any delayed loot drops from any other boss.

I might change that text and add a deathmessage for hierophant to make it seem less confusing and weird.

File: code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -92,7 +92,7 @@
 			var/mob/living/L = user
 			if(!L.can_speak_vocal() || L.oxyloss >= 50)
 				return //stop the sound if oxyloss too high/cant speak
-		playsound(user, user.deathsound, 80, 1, 1)
+		playsound(user, user.deathsound, 200, 1, 1)
```
Comment (ShizCalev) [inline code/modules/mob/living/emote.dm]: TRUE, TRUE

File: code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm [hunk 2]
```diff
@@ -103,15 +105,17 @@ Difficulty: Hard
 				visible_message("<span class='hierophant'>\"Vitemvw gsqtpixi. Stivexmsrep ijjmgmirgc gsqtvsqmwih.\"</span>")
 
 /mob/living/simple_animal/hostile/megafauna/hierophant/death()
-	if(health > 0 || stat == DEAD)
+	if(health > 0 || hierophant_dying)
 		return
 	else
-		stat = DEAD
+		hierophant_dying = 1
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm]: TRUE

File: code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm [hunk 3]
```diff
@@ -75,6 +75,8 @@ Difficulty: Hard
 	var/did_reset = TRUE //if we timed out, returned to our beacon, and healed some
 	var/list/kill_phrases = list("Wsyvgi sj irivkc xettih. Vitemvmrk...", "Irivkc wsyvgi jsyrh. Vitemvmrk...", "Jyip jsyrh. Egxmzexmrk vitemv gcgpiw...", "Kix fiex. Liepmrk...")
 	var/list/target_phrases = list("Xevkix psgexih.", "Iriqc jsyrh.", "Eguymvih xevkix.")
+	var/list/stored_nearby = list() // stores people nearby the hierophant when it enters the death animation
+	var/hierophant_dying = 0 // if we're in the death animation and don't want to proc death repeatedly every time we take damage, do not set stat = dead or deathsounds and messages wont work
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm]: FALSE

**Pull request merged**